snipe/snipe-it
# Fixed missing password.token string and checked for user existing before attempting to send reset email

This fixes a submission from [Huntr.dev](https://huntr.dev/bounties/5c51bfe9-a792-40b5-8b39-7525e93e66b8/) by @imlonghao where the password "token expired" language string was missing and the system would crash if the user didn't exist. This feels like a regression to me, since the comments in the code indicate this was already considered.

To test this before this is merged:

- Open /password/reset/1, for example, https://demo.snipeitapp.com/password/reset/1
- You can see the different responses by using a different username, admin and not-existed-user (make sure you enter a password that match the security rules)
- For the admin user (which DOES exist on the demo), the response is `passwords.token`. (which is missing lang translation I think)
- As for the others, the response is 500 | SERVER ERROR.

An attacker can write a script to detect what users are registered on this snipe-it instance.

### New Flow - INVALID USER

<img width="408" alt="Screen Shot 2022-06-21 at 3 26 03 PM" src="https://user-images.githubusercontent.com/197404/174907110-6f42410f-682c-45db-afce-5fde0a4367df.png">
<img width="428" alt="Screen Shot 2022-06-21 at 3 26 11 PM" src="https://user-images.githubusercontent.com/197404/174907115-37a00874-9602-49d3-8b49-babd2cde20c1.png">

(No email is sent)

### New Flow - VALID USER, INVALID RESET TOKEN (`/password/reset/`)

<img width="614" alt="Screen Shot 2022-06-21 at 3 28 07 PM" src="https://user-images.githubusercontent.com/197404/174907261-b8bca0e5-f5ef-43de-9254-1650e5c96bcc.png">

(No email is sent)

### New Flow - VALID USER, VALID RESET TOKEN

<img width="1462" alt="Screen Shot 2022-06-21 at 4 12 33 PM" src="https://user-images.githubusercontent.com/197404/174911721-0a78b225-3861-455f-8d19-712fabc7cd8f.png">

### Debug Logs

<img width="1410" alt="Screen Shot 2022-06-21 at 4 14 39 PM" src="https://user-images.githubusercontent.com/197404/174911839-3f780e08-5eb3-4c77-992b-20dc5737c771.png">
